File: app/src/main/java/art/coded/wireframe/view/fragment/PlaceholderFragment.java
package art.coded.wireframe.view.fragment;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProvider;

import art.coded.wireframe.databinding.FragmentPlaceholderBinding;
import art.coded.wireframe.viewmodel.PlaceholderViewModel;

public class PlaceholderFragment extends Fragment {

    static final String LOG_TAG = PlaceholderFragment.class.getSimpleName();

    private PlaceholderViewModel placeholderViewModel;
    private FragmentPlaceholderBinding binding;

    public View onCreateView(@NonNull LayoutInflater inflater,
                             ViewGroup container, Bundle savedInstanceState) {
        placeholderViewModel =
                new ViewModelProvider(this).get(PlaceholderViewModel.class);

        binding = FragmentPlaceholderBinding.inflate(inflater, container, false);
        View root = binding.getRoot();

        final TextView textView = binding.textPlaceholder;
        placeholderViewModel.getText().observe(getViewLifecycleOwner(), o -> textView.setText(o));
        return root;
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        binding = null;
    }
}

File: app/src/ktx/java/art/coded/wireframe/view/fragment/PlaceholderFragment.kt
package art.coded.wireframe.view.fragment

import android.view.LayoutInflater
import android.view.ViewGroup
import android.os.Bundle
import androidx.lifecycle.ViewModelProvider
import art.coded.wireframe.viewmodel.PlaceholderViewModel
import android.view.View
import androidx.fragment.app.Fragment
import art.coded.wireframe.databinding.FragmentPlaceholderBinding

class PlaceholderFragment : Fragment() {
    private var placeholderViewModel: PlaceholderViewModel? = null
    private var binding: FragmentPlaceholderBinding? = null
    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?, savedInstanceState: Bundle?
    ): View? {
        placeholderViewModel = ViewModelProvider(this).get(
            PlaceholderViewModel::class.java
        )
        binding = FragmentPlaceholderBinding.inflate(inflater, container, false)
        val root: View = binding!!.root
        val textView = binding!!.textPlaceholder
        placeholderViewModel!!.text
            .observe(viewLifecycleOwner) { o: String? -> textView.text = o }
        return root
    }

    override fun onDestroyView() {
        super.onDestroyView()
        binding = null
    }

    companion object {
        val LOG_TAG = PlaceholderFragment::class.java.simpleName
    }
}